File: Algoritmi/2017-07-31/seats/att/seats.c
#include <stdio.h>
#include <assert.h>

#define MAXQ 100000

int experiment(int N, int Q, char T[], int K[]) {
    // insert your code here
    return 42;
}

char T[MAXQ];
int K[MAXQ];

int main() {
    FILE *fr, *fw;
    int N, Q, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &N, &Q));
    for(i=0; i<Q; i++) {
        assert(1 == fscanf(fr, " %c", &T[i]));
        assert(1 == fscanf(fr, " %d", &K[i]));
    }

    fprintf(fw, "%d\n", experiment(N, Q, T, K));
    fclose(fr);
    fclose(fw);
    return 0;
}

File: Algoritmi/2017-07-31/seats/att/seats.pas
const
    MAXQ = 100000;

function experiment(N, Q: longint; var T: array of char; var K: array of longint): longint;
begin
    (* insert your code here *)
    experiment := 42;
end;


var
    N,Q,i : longint;
    T     : array[0..MAXQ-1] of char;
    K     : array[0..MAXQ-1] of longint;
    fr, fw: text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, Q);
    for i:=0 to Q-1 do begin
        readln(fr, T[i], K[i]);
    end;

    writeln(fw, experiment(N, Q, T, K));
    close(fr);
    close(fw);
end.
